src/services/refactors/generateGetAccessorAndSetAccessor.ts

```diff
@@ -0,0 +1,165 @@
+        const { file, startPosition } = context;
+
+        const fieldInfo = getConvertibleFieldAtPosition(file, startPosition);
+        if (!fieldInfo) return undefined;
+
+        const changeTracker = textChanges.ChangeTracker.fromContext(context);
+        const newLineCharacter = getNewLineOrDefaultFromHost(context.host, context.formatContext.options);
+
+        const { fieldName, accessorName, accessorType, propertyDeclaration, needUpdateName, hasModifiers, needUpdateModifiers } = fieldInfo;
+        const accessorModifiers = hasModifiers ? (
+            (getModifierFlags(propertyDeclaration) & ModifierFlags.Private || !propertyDeclaration.modifiers) ? createNodeArray([createToken(SyntaxKind.PublicKeyword)]) : propertyDeclaration.modifiers
+        ) : undefined;
+
+        const getAccessor = generateGetAccessor(fieldName, accessorName, accessorType, accessorModifiers);
+        const setAccessor = generateSetAccessor(fieldName, accessorName, accessorType, accessorModifiers);
+
+        const modifiers = hasModifiers ? createNodeArray([createToken(SyntaxKind.PrivateKeyword)]) : undefined;
+        if (needUpdateName || needUpdateModifiers) {
+            changeTracker.replaceNode(file, propertyDeclaration, updateoriginalPropertyDeclaration(propertyDeclaration, fieldName, modifiers), {
+                suffix: newLineCharacter
+            });
+        }
+
+        changeTracker.insertNodeAfter(file, propertyDeclaration, getAccessor);
+        changeTracker.insertNodeAfter(file, propertyDeclaration, setAccessor);
+
+        return {
+            edits: changeTracker.getChanges(),
+            renameFilename: undefined,
+            renameLocation: undefined,
+        };
+    }
+
+    function getConvertibleFieldAtPosition(file: SourceFile, startPosition: number): Info | undefined {
+        const node = getTokenAtPosition(file, startPosition, /*includeJsDocComment*/ false);
+        const propertyDeclaration = findAncestor(node.parent, isPropertyDeclaration);
+
+        if (!propertyDeclaration || propertyDeclaration.name.kind !== SyntaxKind.Identifier) return undefined;
+        // make sure propertyDeclaration have only AccessibilityModifier
+        if ((getModifierFlags(propertyDeclaration) | ModifierFlags.AccessibilityModifier) !== ModifierFlags.AccessibilityModifier) return undefined;
+
+        const containerClass = getContainingClass(propertyDeclaration);
+        if (!containerClass) return undefined;
+
+        const members = getMembersOfDeclaration(containerClass);
+        if (!members) return undefined;
+
+        const needUpdateName = propertyDeclaration.name.text.charCodeAt(0) !== CharacterCodes._;
+
+        const accessorName = needUpdateName ? propertyDeclaration.name.text : propertyDeclaration.name.text.substring(1);
+        const fieldName = `_${accessorName}`;
+
+        if (find(members, member => needUpdateName ? member.name.getText() === fieldName : member.name.getText() === accessorName)) return undefined;

```

Comment:
Prefer `members.some(...)` over `find` when just using as a boolean

Reply:
Also, `.getText()` on every member might be slow since `getText()` has to do scanning. Might want to do something like `function getDeclarationName` from `services.ts` to just get Identifier and StringLiteral-named properties.

Reply:
Also, `getMemberName(member.name) === (needUpdateName ? fieldName : accessorName)`

Reply:
Actually, can check for `checker.getPropertyOfType(checker.getTypeAtLocation(containerClass.name), needUpdateName ? fieldName : accessorName)` which avoids conflicts with supertypes too.